src/renderer.c

```diff
@@ -215,54 +208,41 @@ static void font_clear_glyph_cache(RenFont* font) {
   }
 }
 
+// based on https://github.com/libsdl-org/SDL_ttf/blob/2a094959055fba09f7deed6e1ffeb986188982ae/SDL_ttf.c#L1735
+static unsigned long font_file_read(FT_Stream stream, unsigned long offset, unsigned char *buffer, unsigned long count) {
+  Sint64 amount;
+  SDL_RWops *file = (SDL_RWops *) stream->descriptor.pointer;
+  SDL_RWseek(file, (int) offset, RW_SEEK_SET);
+  if (count == 0)
+    return 0;
+  amount = SDL_RWread(file, buffer, sizeof(char), count);
+  if (amount <= 0)
+    return 0;
+  return (unsigned long) amount;
+}
+
 RenFont* ren_font_load(RenWindow *window_renderer, const char* path, float size, ERenFontAntialiasing antialiasing, ERenFontHinting hinting, unsigned char style) {
+  RenFont *font = NULL;
   FT_Face face = NULL;
-
-#ifdef _WIN32
-
-  HANDLE file = INVALID_HANDLE_VALUE;
-  DWORD read;
-  int font_file_len = 0;
-  unsigned char *font_file = NULL;
-  wchar_t *wpath = NULL;
-
-  if ((wpath = utfconv_utf8towc(path)) == NULL)
-    return NULL;
-
-  if ((file = CreateFileW(wpath,
-                          GENERIC_READ,
-                          FILE_SHARE_READ, // or else we can't copy fonts
-                          NULL,
-                          OPEN_EXISTING,
-                          FILE_ATTRIBUTE_NORMAL,
-                          NULL)) == INVALID_HANDLE_VALUE)
-    goto failure;
-
-  if ((font_file_len = GetFileSize(file, NULL)) == INVALID_FILE_SIZE)
-    goto failure;
-
-  font_file = check_alloc(malloc(font_file_len * sizeof(unsigned char)));
-  if (!ReadFile(file, font_file, font_file_len, &read, NULL) || read != font_file_len)
+  
+  SDL_RWops *file = SDL_RWFromFile(path, "rb");
+  if (!file)
     goto failure;
 
-  free(wpath);
-  wpath = NULL;
+  int len = strlen(path);
+  font = check_alloc(calloc(1, sizeof(RenFont) + len + 1));
+  font->stream.read = font_file_read;
```

Comment:
Should we also define a `close` function?

Reply:
I forgot why I didn't do that, but I think its because there's something weird about it. I'll come back to it.